### DISCUSSION TITLE
"Read data in SQL console" should not create dirty editor

### DISCUSSION THREAD

**Discussion by 2019-05-10** [dbeaver/dbeaver#11232] [Q&A] [answered]
On a table rmc -> "Read data in SQL console" opens an editor with the SQL query executed.
Closing the editor w/o any change nevertheless prompts "Save changes in Script-XXX.sql?"

Since the editor's contents
a) are automatically generated and thus always the same
b) have not manually changed
the prompt is unnecessary, and in respect to b) unjustified, and should not appear at all.

**Comment by serge-rider** [comment, marked as answer]
Hint: don't use "Read data in SQL console" if you need to read table data only. Use "View data" action or just double-click on a table and switch to "Data" tab.

"Read data in SQL console" is supposed to be used when you are planning to change the query manually.
